plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotIntegrationSmokeTest.java

```diff
@@ -0,0 +1,386 @@
+                .name("float_array_col").type().optional().array().items().nullable().floatType()
+                .name("double_array_col").type().optional().array().items().nullable().doubleType()
+                .name("long_array_col").type().optional().array().items().nullable().longType()
+                .name("int_col").type().optional().intType()
+                .name("float_col").type().optional().floatType()
+                .name("double_col").type().optional().doubleType()
+                .name("long_col").type().optional().longType()
+                .name("updated_at").type().optional().longType()
+                .endRecord();
+    }
+
+    private static String getMixedCaseValues(List<List<Object>> resultRows, UnaryOperator<List<Object>> transform)
+    {
+        checkState(!resultRows.isEmpty(), "resultRows is empty");
+        StringBuilder valuesBuilder = new StringBuilder("VALUES ");
+        return valuesBuilder.append(String.join(", ", resultRows.stream()
+                .map(transform)
+                .map(row -> "(" + String.join(", ", row.stream().map(column -> "'" + column + "'").collect(toList())) + ")")
+                .collect(toList()))).toString();
+    }
+
+    @Test
+    public void testMixedCase()
+    {
+        UnaryOperator<List<Object>> transform = resultRow ->
+                ImmutableList.builder()
+                        .add(resultRow.get(0))
+                        .add(resultRow.get(1))
+                        .add(TimeUnit.SECONDS.convert((long) resultRow.get(2), TimeUnit.MILLISECONDS))
+                        .build();
+
+        assertQuery("SELECT stringcol, longcol, updatedatseconds FROM " + MIXED_CASE_TABLE + " LIMIT " + MAX_ROWS_PER_SPLIT, getMixedCaseValues(MIXED_CASE_RESULTS.subList(0, 1000), transform));
+        assertQuery("SELECT stringcol, longcol, updatedatseconds FROM" +
+                "  \"SELECT updatedatseconds, longcol, stringcol FROM " + MIXED_CASE_TABLE + " LIMIT " + MAX_ROWS_PER_SPLIT + "\"",
+                getMixedCaseValues(MIXED_CASE_RESULTS.subList(0, 1000), transform));
+        assertQuery("SELECT stringcol, longcol, updatedatseconds FROM " + MIXED_CASE_TABLE + " WHERE longcol = 3", getMixedCaseValues(MIXED_CASE_RESULTS.stream()
+                .filter(row -> (long) row.get(1) == 3)
+                .collect(toList()),
+                transform));
+        assertQuery("SELECT stringcol, longcol, updatedatseconds FROM" +
+                "  \"SELECT updatedatseconds, longcol, stringcol FROM " + MIXED_CASE_TABLE + "\" WHERE longcol = 3", getMixedCaseValues(MIXED_CASE_RESULTS.stream()
+                .filter(row -> (long) row.get(1) == 3)
+                .collect(toList()),
+                transform));
+    }
+
+    @Test
+    public void testLimitForSegmentQueries()
+    {
+        assertQueryFails("SELECT stringcol, longcol, updatedatseconds FROM " + MIXED_CASE_TABLE, "Segment query returned.*");
```

Comment:
Expanded the message. The test is expected to fail because we set the `pinot.max-rows-per-split-for-segment-queries` to 1000. If a segment query (directly from the pinot server, not the pinot broker) returns > 1000 rows the query fails.

The reason for this limit is that if too many rows (i.e. select w/o filter or limit) are returned directly from the servers they can crash. Pinot is designed to work well with aggregate queries, where the broker returns the results (using pre aggregated indexes, etc.), rather than the server returning the results and the trino engine processing them.

Reply:
Context: "segment" query fetches the data from the servers which store the data, does not take advantage of preaggregated indexes (in pinot called "star tree" indexes). A "broker" query will usually fetch far less data from the servers because it can take advantage of the indexing. Sometimes segment queries are unavoidable, if some parts of the query can not be pushed down.